### ISSUE TITLE
Bash Support For Windows

### ISSUE THREAD

**Issue by fkorotkov** [cirruslabs/cirrus-ci-docs#408]
# Description

Right now setting up `CIRRUS_SHELL` to `bash` for `windows_container`s doesn't work. It should be supported to images with bash installed.

# Context

https://twitter.com/DanTup/status/1158733530479890437

**Comment by fkorotkov** [comment]
@DanTup do you happen to know why executing

```bash
C:\tools\msys64\usr\bin\bash -lc 'printenv'
```

Succeeds but having `printenv` in a bash  file `foo.sh` leads to an [error](https://cirrus-ci.com/task/5844826593951744?command=printenv#L1) when  running `C:\tools\msys64\usr\bin\bash foo.sh`:

```bash
printenv
C:\Users\ContainerAdministrator\AppData\Local\Temp\scripts52fdfc072182654f163f5f0f9a621d72.sh: line 4: printenv: command not found
Exit status: 127
```

My Windows knowledge are very limited. Was wondering if you might experience something similar.

**Comment by RDIL** [comment]
maybe the `PATH`s are different

**Comment by DanTup** [comment]
Hmm, I'm, not sure - I've used bash on Windows in Travis, but never had to set it up :D

It does seem like `PATH` may be wrong, but I couldn't explain why. You may be able to verify that by doing `echo $PATH` to see if it differs between the two. I'm away from home so have no Windows access, but I can do some testing of this when I'm back if you don't figure it out.

Some stabs in the dark - does running `sh foo.sh` differ, or adding `#!/usr/bin/env sh` or similar to the top of `foo.sh`?

**Comment by DanTup** [comment]
I just tested this and saw the same. Putting `/usr/bin/printenv` does work, however.

It seems that the `-lc` command has an updated PATH. It has this added to the start:

`/c/Users/danny/bin:/usr/local/bin:/usr/bin:/bin:/opt/bin`

and this added to the end:

`:/usr/bin/vendor_perl:/usr/bin/core_perl`

If you run the script also with with the `-l` flag, that also fixes it - maybe that's telling it to do something that sets up the path (run login scripts? I can't find any docs).

TL;DR: Use `C:\tools\msys64\usr\bin\bash -l foo.sh` instead :)

**Comment by k0ekk0ek** [comment]
The agent supports `bash` already:

https://github.com/cirruslabs/cirrus-ci-agent/blob/fa304ac83aed82f41558cd9141fe3095cf13889d/internal/executor/cmd_windows.go#L43-L61

I managed to use a `bash` shell on a `windows_container` without any problems. The trick is to add: `C:\Program Files\Git\bin` and `C:\Program Files\Git\mingw64\bin` to the `PATH` environment variable first. I did so by using the `dockerfile:` approach, but if the base image were to be updated to include it by default everything works out of the box if you supply `CIRRUS_SHELL: bash`.

**Comment by fkorotkov** [comment]
@k0ekk0ek thank you for the update! will close the issue then.
